### PR TITLE
Adding PDB chainID support

### PR DESCRIPTION
**Pull request by pgbarletta**
From #873  and #1714 .

### CODE REVIEW

**Comment by pgbarletta** [comment]
I hadn't run `test_trajectory.py` on my system, but yeah, it seems those tests are failing.

File: mdtraj/core/topology.py [hunk 1]
```diff
@@ -1295,13 +1295,15 @@ class Chain(object):
...
-    def __init__(self, index, topology):
+    def __init__(self, index, topology, chain_id = None):
         """Construct a new Chain.  You should call add_chain() on the Topology instead of calling this directly."""
         # The index of the Chain within its Topology
         self.index = index
         # The Topology this Chain belongs to
         self.topology = topology
         self._residues = []
+        # PDB format chainID
+        self.chain_id = chain_id
```
Comment (sroet) [inline mdtraj/core/topology.py:1306]: please add a description of this attribute to the docstring above

File: mdtraj/core/topology.py [hunk 2]
```diff
@@ -1295,13 +1295,15 @@ class Chain(object):
         Iterator over all Atoms in the Chain.
     """
 
-    def __init__(self, index, topology):
+    def __init__(self, index, topology, chain_id = None):
```
Comment (sroet) [inline mdtraj/core/topology.py:1298]: [pep8](https://www.python.org/dev/peps/pep-0008/) style:
```suggestion
    def __init__(self, index, topology, chain_id=None):
```

File: mdtraj/core/topology.py [hunk 3]
```diff
@@ -636,15 +636,15 @@ def __eq__(self, other):
 
         return True
 
-    def add_chain(self):
+    def add_chain(self, chain_id = None):
```
Comment (sroet) [inline mdtraj/core/topology.py:639]: [pep8](https://www.python.org/dev/peps/pep-0008/) suggestion
```suggestion
    def add_chain(self, chain_id=None):
```

File: mdtraj/formats/pdb/pdbfile.py [hunk 4]
```diff
@@ -332,7 +332,10 @@ def write(self, positions, topology, modelIndex=None, unitcell_lengths=None,
         if modelIndex is not None:
             print("MODEL     %4d" % modelIndex, file=self._file)
         for (chainIndex, chain) in enumerate(topology.chains):
-            chainName = self._chain_names[chainIndex % len(self._chain_names)]
+            if chain.chain_id == None or chain.chain_id == '':
+                chainName = self._chain_names[chainIndex % len(self._chain_names)]
+            else:
+                chainName = chain.chain_id[0:1]
```
Comment (sroet) [inline mdtraj/formats/pdb/pdbfile.py:338]: can be simplified as both `None` and `''` evaluate to `False`
```suggestion
            if not chain.chain_id:
                chainName = self._chain_names[chainIndex % len(self._chain_names)]
            else:
                chainName = chain.chain_id[0:1]
```

**Review by sroet — CHANGES_REQUESTED** [review]
Some code style suggestions.

Also, the test error seems genuine (missing the added `chain_id` from the `Parameters` section in the docstring of that class):
```
ValueError: In mdtraj.core.topology.add_chain, arguments ['chain_id'] don't match Parameters list []
```

File: mdtraj/core/topology.py (hunk 1 above)
Comment (pgbarletta) [reply mdtraj/core/topology.py:1313]: done

File: mdtraj/core/topology.py (hunk 2 above)
Comment (pgbarletta) [reply mdtraj/core/topology.py]: funny, my pep8 was messing with the files and I had to turn it off. Done.

File: mdtraj/core/topology.py (hunk 3 above)
Comment (pgbarletta) [reply mdtraj/core/topology.py]: Done.

File: mdtraj/formats/pdb/pdbfile.py (hunk 4 above)
Comment (pgbarletta) [reply mdtraj/formats/pdb/pdbfile.py]: you're right, thanks. Done.

**Review by rmcgibbo — APPROVED** [review]

**Comment by rmcgibbo** [comment]
@sroet: any more changes from you?

**Review by sroet — APPROVED** [review]
looks good to me

**Pull request merged**